snuba/datasets/storages/transactions.py

```diff
@@ -94,13 +94,11 @@
     storage_set_key=StorageSetKey.TRANSACTIONS,
     schema=schema,
     query_processors=[
+        MappingColumnPromoter(mapping_specs={"contexts": {"trace_id": "trace_id"}}),
```

Comment:
Wait how does this work ? The `MappingColumnPromoter` does not take the type of the tag/context into account and transforms `valueOf(contexts.value, indexOf(context.keys, 'trace_id'))` into `trace_id`. But it does not do anything about the fact that the context in the trace ctx is actually not formatted as a UUID. While the `trace_id` column is of UUID type. 
I think making this change would make us loose what we did with TypedContextPromoter

Reply:
It works because the trace_id column is converted to the right type by the UUIDColumnProcessor https://github.com/getsentry/snuba/blob/master/snuba/datasets/storages/transactions.py#L107

Reply:
ok, can we move the UUIDColumnProcessor up, right after this one? So it would make it clear that this resolves a context to a column and the UUIDColumnProcessor fixes the type for what this one modified.
Also What are you planning to do about the span_id? It feels weird that we have one approach to promote UUID fields and a different one for Hex fields.

Reply:
Moved up. span_id / HexInt is next on my list to convert to a similar approach.